tips/TIP-0033/tip-0033.md

```diff
@@ -0,0 +1,201 @@
+
+An open-source and publicly maintained registry, the system provides a portal for projects to register their tokens to maintain a publicly accessible reference for dApp providers and users to validate official project tokens.
+
+The registry comprises of:
+- **iNFT Policy Registry** to allow open-source verifiable collection details
+- **IOTA Native Token Registry** to allow open-source verifiable token details
+- **REST API** to allow easy integrations into dApps
+- **Registry Front-End** to provide an accessible registry search that can be added to any application
+- **GitHub PR Bot** to provide easily accessible registry applications
+
+Utilising the schema from **IRC27 Standards for iNFTs** and **IRC30 Standards for IOTA Native Tokens**, supported with easily deployable end-points and APIs, the registry provides an accessible integration to build trust and security within the IOTA tokenisation ecosystem.
+
+
+## Motivation
+
+
+### Why a Public Registry?
+
+With an expanding decentralised ecosystem the number of tokens, digital assets, and NFT collections will grow at an extremely high rate. Keeping track of authentic sources is a continued struggle, with opportunists often building out potentially malicious clone tokens and NFT collections in order to take advantage of purchasers through deceptive means.
+
+A universal registry built not only for managing Native Tokens, but also for the management of verified NFT Policy IDs, provides an open-source point of trust and validation, that can enable creators and users to easily verify the authenticity of a project's token or assets through a simple API call or live reference on the public registry.
+
+This document we will present the deployment and maintenance of the **IOTA Public Token Registry**.
+
+
+## Specification
+
+### The Registry
+
+An open source registry with files containing JSON(s) for registered native tokens and NFT collections that can support project creators in opening a Pull Request to add their token to the registry.
+
+The registry repository will follow the below folder structure whereby users can submit a JSON file with their project schema and wait for it to be approved by the committee of verifiers.
+
+
+```
+   repository 
+       |
+       |__ tokens 
+       |     |__ sample-token-project-1.json
+       |     |__ sample-token-project-2.json
+       |
+       |-- NFTs
+             |__ sample-nft-project-1.json
+             |__ sample-nft-project-2.json
+```
+
+The registry will be moderated and maintained by a Committee of Human validators whose approval is needed before anything get's merged. The Committee itself will be a operated via a DAO, the DAO will control who get's to be a part of the approval committee and members can be removed or added by the DAO.
```

Comment:
This feels way to centralized and manual. Do we really want to have "Human validators" for this?

How do you make sure this scale, it does not hurt adoption and it's decentralized? Especially decentralization aspect needs way more details and you can't rely on just one approach. 

I understand this might be an approach by Cardano but they themselves are experiences many issues by just looking at their repo. See here: 
- https://github.com/Cardano-NFTs/policyIDs/issues/6744
- https://github.com/Cardano-NFTs/policyIDs/issues/6649
- look how many pending PR requests they have

I understand this approach and surely can work for some DAOs but it should not be enforced as primary method by IOTA (single point of record). IOTA should provide various approaches and users will decide the one they want to trust.

Reply:
I don't think the authors mean this registry as the ultimate solution, it's rather just a first step.

In the DLT world, on-chain assets are always identified by random numbers (hashes, IDs, etc.) To associate these long random numbers that humans don't ever want to see with meaningful names, there always has to be social consensus on which random number maps to what name.

At the end of the day, real world idenities must be linked to on-chain indentities. If we had all the on-chain addresses of NFT issuers published and verified off-chain, we could easily associate IRC-27 NFTs with their verified issuers, hence we can trust that the asset is legitimate. The real question is: how do you do the linking?

There are several ways:
 1. Leave it up to the user to dig up the information on the internet and decide based on the found info if they trust it or not.
 2. Have a 3rd party being responsible to do the verification off-chain, and supply the info to end users. Like twitter verified profiles.
 3. Have the community delegate experts who come together and do verification based on majority decision in an open and transparent way.
 4. Make the verification process more decentralized by having token holders vote on to-be submitted entries.
 5. Make the verification process completely automatic and decentralized by having an SC giving out verified pledges based on deterministic information. This also means that there is no social consensus, I can just buy the name `CocaColaNFT` even if I have no association whatsoever with Coca Cola.
 
All 5 are possible, but they each have their constraints. IOTA (and Shimmer) will not enforce any of these on protocol level. Applications, wallets, users should have a chance to choose which verifier service to use. TIP-27 is about what format to use to link data to an NFT, not if the data itself is legit or not.
 
[ CFNT](https://cnft.io/) is a marketplace on cardano that would fall into category `2`. There is also [Cardano Token Registry](https://github.com/cardano-foundation/cardano-token-registry) which does the same, but the trust is in the Cardano Foundation.

Based on TIP-33 my understanding is that we want to have a token registry in category `3`. Of course, it doesn't sclae well (nothing does that include social consensus), but you can trust it at least that the majority of community experts did their job with verification, there is no single point of failure.

It can also transition to `4`, or `5` with a limited set of reserved names that do require social consensus.

What do you think? How does it work in other NFT systems? Don't you have 3rd parties like opensea who verify that the issuers and the issued tokens are verified?

Reply:
What Levente said :D

Yes, this is not the final solution, but as an accessible primary starting point it is a viable solution and a very easily to implement, functional  deployment. In regard to the CNFT as an example, this was purely from a technical oversight, that repository is managed by a very limited dev team of just one or two validators, hence its sometimes slow pace at PR confirmations. 

With a well defined and structured decentralised committee of DAO members responsible for the validation in the IOTA ecosystem, this would be less of a concern. There are plenty of community individuals that would be more than willing to participate and support the registry. 

The main issue when you run into no.4 is this increases the delay in the verifiacation process, and with 5 how could an automated system be deployed to control IP registrations and copycat projects? If an individual is first in with $SOON token or the Soonaverse collection, it is verified and added to the registry, which I am sure wouldn't be agreeable for the Soonaverse team. With an early implementation of a decentralised committee of human validators we can negate these attack vectors with peer reviewed regitration applications.

Reply:
So I’ve based on my opinion on what’s in the specification. The specification implies to the reader there will be only one system and that’s TIP-33.

It says:

> 
> To call the defined `policyId` you should request the `policyId` UTXO of the collection NFT and resolve to the Policy > Registry for human identifiable verification.
> 
> To better serve the ecosystem with a single point of record for registered collections a Public Token Registry is defined in [IOTA Public Token Registry - TIP 33](https://github.com/iotaledger/tips/pull/--) where further reading can be found.

I completely agree with above @lzpap and yes there are several systems that exists. I’m not trying to argue which one works which one not. How policy id works, etc…It might not have been very clear in my previous post. Although, I don’t think IOTA should be encouraging just one system (i.e. TIP-33) which you seem to do. It should rather focus on an open protocol and leave various policies to be implemented by third parties.

Reply:
There is always the potential to add future TIPs with future policies and standards. All standards and proposals are individual, there are many more standards and registries than just ERC-721 and the Ethereum Foundation Token Registry, for example. These are simply recommended first iterations, which will, in time, expand and no doubt be added further too - hence the inclusion of versioning for backwards compatibility. 

I also agree this is not the final solution, but as a starting point it makes for an easy to adopt ecosystem wide interoperable solution that can really support security, accessibility, and ease of adoption across multiple dApps from day one.

Reply:
That's fine. 

But again, this is not what the specification tells us. I'm very happy with you proposing TIP-33. I'm sure it'll work in some scenarios. But in my opinion it should not be mentioned in TIP-27. TIP-27 should only talk about framework that will have a support for various registries, identities and so on. It should not be pointing to just "one solution". It should not have preferred initial method. That's how it reads at the moment.

I've bounced the idea with few well established artists and they've not liked it. Have you done any analysis with the ecosystem? 

For example as a side note, Soonaverse have decided that we should not do any moderation neither try to define it as it would put is in conflict and make us rather centralized. We told community to decide on the best approaches and we will make sure to give them tools they need. We've got a specific channel on our discord (https://discord.com/channels/892481940953915433/954144769515335720). Please read this and you would notice that it's not as simple as having a moderated registry. Community is clearly looking for different solutions. 

Therefore, I believe it's safer for IF to just focus on providing a generic framework and keep TIP-33 as independent proposal that is not mixed with TIP-27.

Reply:
Okay, that's fine. I see your point.